source/common/router/header_formatter.cc

```diff
@@ -143,7 +143,10 @@ RequestInfoHeaderFormatter::RequestInfoHeaderFormatter(absl::string_view field_n
     }
     field_extractor_ = [this, pattern](const Envoy::RequestInfo::RequestInfo& request_info) {
       const auto& formatters = start_time_formatters_.at(pattern);
-      ASSERT(formatters.size() == 1);
+      if (formatters.size() != 1) {
```

Comment:
@dio does this seem right?

Reply:
On the second thought, I think we can support multiple patterns. 

@anirudhmurali Do you want to take it?